storm-client/src/jvm/org/apache/storm/drpc/DRPCSpout.java

```diff
@@ -105,13 +105,19 @@ public Adder(String server, int port, Map<String, Object> conf) {
 
         @Override
         public Void call() throws Exception {
-            DRPCInvocationsClient c = new DRPCInvocationsClient(conf, server, port);
-            synchronized (_clients) {
-                _clients.add(c);
+            try{
+                DRPCInvocationsClient c = new DRPCInvocationsClient(conf, server, port);
+                synchronized (_clients) {
+                    _clients.add(c);
+                }
+            }catch (Exception e){
+                LOG.warn("Can't connect to some drpcserver when init drpcspout,please check your cluster.");
```

Comment:
If we are going to warn can we at least log what the exception was that made it so we cannot connect?